pkg/plugin/query.go

```diff
@@ -26,7 +26,11 @@ func (h *Handler) QueryData(ctx context.Context, req *backend.QueryDataRequest)
 			continue
 		}
 
-		res, err := h.QueryRunner.Run(qr)
+		authHeaders := humio.AuthHeaders{
+			"Authorization": req.GetHTTPHeader("Authorization"),
+			"X-Id-Token":    req.GetHTTPHeader("X-Id-Token"),
+		}
+		res, err := h.QueryRunner.Run(qr, authHeaders)
```

Comment:
You could add auth headers to the client, and then you could set the headers here. i think that gets around passing the headers as params